sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1827,3 +1825,43 @@ def test_scores_attribute_layout_elasticnet():
 
             avg_score_lr = cross_val_score(lr, X, y, cv=cv).mean()
             assert avg_scores_lrcv[i, j] == pytest.approx(avg_score_lr)
+
+
+@pytest.mark.parametrize("fit_intercept", [False, True])
+def test_multinomial_identifiability_on_iris(fit_intercept):
+    """Test that the multinomial classification is identifiable.
+
+    A multinomial with c classes can be modeled with
+    probability_k = exp(X@coef_k) / sum(exp(X@coef_l), l=1..c) for k=1..c.
+    This is not identifiable, unless one chooses a further constraint.
+    According to [1], the maximum of the L2 penalized likelihood automatically
+    satisfies the symmetric constraint:
+    sum(coef_k, k=1..c) = 0
+
+    Further details can be found in the appendix of [2].
+
+    Reference
+    ---------
+    .. [1] Zhu, Ji and Trevor J. Hastie. "Classification of gene microarrays by
+    penalized logistic regression". Biostatistics 5 3 (2004): 427-43.
+    https://doi.org/10.1093/biostatistics%2Fkxg046
+
+    .. [2] Powers, Scott, Trevor J. Hastie and Robert Tibshirani. "Nuclear
+    penalized multinomial regression with an application to predicting at bat
+    outcomes in baseball." Statistical modelling 18 5-6 (2017): 388-410 .
+    https://arxiv.org/pdf/1706.10272.pdf
+    """
+    # Test logistic regression with the iris dataset
+    n_samples, n_features = iris.data.shape
+    target = iris.target_names[iris.target]
+
+    clf = LogisticRegression(C=len(iris.data), solver='lbfgs', max_iter=300,
```

Comment:
If this is always the case, does it make sense to run this test for random datasets with `make_classification()` with different `random_state`s?

Reply:
Yes, this test should hold for every LR on multiclass datasets. Do you want me to change iris to `make_classification`?

Reply:
Yea, I think using `make_classification` with a pytest parameterize with `random_state=[0, 1, 2]`.

Hopefully this does not use too much time.

Reply:
> run this test for random datasets with make_classification() with different random_state

It's a general issue we have of how to make sure the test holds for different random states. I don't think parametrizing with `random_state` is a solution as then we would start doing that in every other test. IMO verifying  this property of iris should be enough and is already an improvement.

Reply:
Okay, lets keep it only for the iris data set.